extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/kernel/StageDefinition.java

```diff
@@ -74,7 +74,7 @@
  */
 public class StageDefinition
 {
-  private static final int PARTITION_STATS_MAX_KEYS = 2 << 15; // Avoid immediate downsample of single-bucket collectors
+  private static final int PARTITION_STATS_MAX_BYTES = 300_000_000; // Avoid immediate downsample of single-bucket collectors
```

Comment:
Ideally, we should have this limit as the % of the JVM memory passed to the worker. But We can do that change later.

Also, We might want to document this somewhere in the docs. 
https://druid.apache.org/docs/latest/multi-stage-query/concepts.html#memory-usage
`On Peons launched by Middle Managers, the bulk of the JVM heap (75%) is split up into two bundles of equal size: one processor bundle and one worker bundle. Each one comprises 37.5% of the available JVM heap.`
Maybe here.

Reply:
Added